lib/model/content.dart

```diff
@@ -627,6 +630,25 @@ class _ZulipContentParser {
     return ListNode(listStyle!, items, debugHtmlNode: debugHtmlNode);
   }
 
+  // Ported from https://github.com/zulip/zulip-mobile/tree/main/src/emoji/data.js
+  //
+  // Which was in turn ported from https://github.com/zulip/zulip/blob/main/zerver/models.py
+  // and that describes the encoding as follows:
+  //
+  // > * For Unicode emoji, [emoji_code is] a dash-separated hex encoding of
+  // >   the sequence of Unicode codepoints that define this emoji in the
+  // >   Unicode specification.  For examples, see "non_qualified" or
+  // >   "unified" in the following data, with "non_qualified" taking
+  // >   precedence when both present:
+  // >   https://raw.githubusercontent.com/iamcal/emoji-data/master/emoji_pretty.json
+  String? tryParseEmojiCodeToUnicode(String code) {
+    try {
+      return String.fromCharCodes(code.split('-').map((hex) => int.parse(hex, radix: 16)));
+    } catch (_) {
```

Comment:
Yeah, `catch (_)` is a bit of a code smell :-)

Generally if there's some exception we expect to sometimes happen, and doesn't represent a bug in the code, we'll want to be explicit about what we expect and just catch that exception type specifically. That way if some unrelated exception is thrown as a result of a bug, the exception propagates just like it normally would.

Reply:
ack